import bug

1- Create a file src/plugins/plugin.js with code :
export class Plugin {
    constructor() {
        this.callbacks = [];
    }

    onEvent(callback) {
        this.callbacks.push(callback);
    }
}

2 - Add to keyboard and touch index.js plugin file :
import { Plugin } from '@plugins/plugin.js';

Error :
Uncaught (in promise) Error: Cannot find module '../../plugins/plugin.js'

--- a/src/plugins/input/keyboard/index.js
+++ b/src/plugins/input/keyboard/index.js
@@ -1,15 +1,12 @@
+'use strict';
+
 import config from '@config/config.json';
+import { Plugin } from '@plugins/plugin.js';
 
-class KeyboardInput {
+class KeyboardInput extends Plugin {
     constructor() {
-        this.usedByAComponent = true;
+        super();
         this.type = 'inputEvent';
-        this.callbacks = [];
-        this.initialized = false;
-    }
-
-    onEvent(callback) {
-        this.callbacks.push(callback);
     }
 
     init() {

--- a/src/plugins/input/touch-pointer/index.js
+++ b/src/plugins/input/touch-pointer/index.js
@@ -1,17 +1,15 @@
-class TouchPointerInput {
+'use strict';
+
+import { Plugin } from '@plugins/plugin.js';
+
+class TouchPointerInput extends Plugin {
     constructor() {
-        this.usedByAComponent = true;
+        super();
         this.type = 'touchPointerEvent';
-        this.callbacks = [];
         this.zooming = false;
         this.pointer = { x: 0, y: 0, color: '#FF0000' };
         this.interval;
         this.messageEventRegistered = false;
-        this.initialized = false;
-    }
-
-    onEvent(callback) {
-        this.callbacks.push(callback);
     }
 
     init() {

--- a/src/plugins/input/touch/index.js
+++ b/src/plugins/input/touch/index.js
@@ -1,19 +1,16 @@
-import contextService from '@services/context';
+'use strict';
 
-class TouchInput {
+import contextService from '@services/context';
+import { Plugin } from '@plugins/plugin.js';
+
+class TouchInput extends Plugin {
     constructor() {
-        this.usedByAComponent = true;
+        super();
         this.type = 'inputEvent';
         this.touchPosition = {
             touchstart: { clientX: 0, clientY: 0 },
             touchend: { clientX: 0, clientY: 0 }
         };
-        this.callbacks = [];
-        this.initialized = false;
-    }
-
-    onEvent(callback) {
-        this.callbacks.push(callback);
     }
 
     init() {

--- a/src/plugins/plugin.js
+++ b/src/plugins/plugin.js
@@ -1,0 +1,13 @@
+'use strict';
+
+export class Plugin {
+    constructor() {
+        this.usedByAComponent = true;
+        this.initialized = false;
+        this.callbacks = [];
+    }
+
+    onEvent(callback) {
+        this.callbacks.push(callback);
+    }
+}

